public/app/features/search/page/components/columns.tsx

```diff
@@ -442,3 +498,35 @@ function getDisplayValue({
   }
   return formattedValueToString(getDisplay(value));
 }
+
+const MILLISECONDS_IN_MINUTE = 60 * 1000;
+const MINUTES_IN_HOUR = 60;
+const HOURS_IN_DAY = 24;
+
+/**
+ * Calculates the rough duration with between two dates, with varying precision depending on
+ * how far apart the dates are. e.g:
+ *  - 20 days, 13 hours, 43 minutes apart, 35 seconds - returns { days: 20, hours: 0, minutes: 0 }
+ *  - 13 hours, 43 minutes apart, 35 seconds - returns { days: 0, hours: 13, minutes: 0 }
+ *  - 43 minutes apart, 35 seconds - returns { days: 0, hours: 0, minutes: 43 }
+ *  - 35 seconds - returns { days: 0, hours: 0, minutes: 0 }
+ */
+function calcCourseDuration(start: Date, end: Date) {
```

Comment:
Intererested in people's take on this. The goal here is to only show the most significant unit, capped at days.

date-fns does have `intervalToDuration` which does something very similar except:
 - it returns `{ months: 1 }` for items as they're immediately deleted
 - it returns the exact duration `{ months: 1, minutes: 47 }` which is more than we need here

So to use intervalToDuration we would still need to do some futzing with the output to get it into a state we want.

Thoughts?

Reply:
I think this makes sense

Reply:
Can't we just call `intervalToDuration` and then modify the values? Seems a bit clearer, smth like this: 
```ts
function calcCourseDuration(start, end) {
    const fullDuration = intervalToDuration({ start, end });

    let { days = 0, hours = 0, minutes=0 } = fullDuration;

    if (days > 0) {
        hours = 0;
        minutes = 0;
    } else if (hours > 0) {
        minutes = 0;
    } 

    return {
        days,
        hours,
        minutes,
    };
}
```

Reply:
> - it returns { months: 1 } for items as they're immediately deleted

The initial hard delete date is 30 days in the future, which is just enough to trigger `{months: 1, minutes: 48}` to be returned from `intervalToDuration`, so this'll return incorrectly.

Initially I wanted to avoid showing "1 month" in the UI, and instead prefer "30 days", as I thought it was the right level of courseness, but totally happy to revisit that.

If we're happy with showing "1 month", we could add months (and probably years) to something like your above logic, but then we're not too far off from what I had. It avoids actual math I guess.

This is exactly why I wanted people's take on it - am I making the right trade offs/design decisions?

Reply:
Do I understand it right: You would like to get rid of the lines 515-525 and replace it by `intervalToDuration()` plus removing seconds?
If yes, please go ahead.

Reply:
Right so there's a product/design decision to be made. When an dashboard has just been deleted, should it say "1 month" or "30 days"? I had a preference for "30 days", but I'm not set on that. If we're happy with it saying "1 month", then we can use `intervalToDuration`.

Then, for the "rough" time remaining, it's not _just_ removing seconds. It's removing all units smaller than the largest unit.

Consider if the deletion date is slightly over 1 month away, as it frequently is when it's been immediately deleted. The deletion date will be `{ months: 1, minutes: 47, seconds: 13  }`. We don't want it to read "1 month, 47 minutes, 13 seconds" (or "30 days, 47 minutes, 13 seconds") as really the minutes and seconds are insignificant in comparison. So days, minutes, and seconds need to be potentially removed as well.

Reply:
It would basically only say 1 month on the first day? I think that could be acceptable. But I prefer days in general.

Reply:
I agree that showing `30 days` is a bit better, but `1 month` is also fine imo.

Reply:
soft agree with alex, i think the difference between `1 month` and `30 days` is minor and we should do what's simplest code-wise here